### ISSUE TITLE
"Keep the same fingering..." option to be added to transposition feature

### ISSUE THREAD

**Issue by abariska** [musescore/MuseScore#17820]
### Your idea

In current master of MU4, when I use transposition for the Tab staff (separated or linked) notes on the tab staff ae optimised and can be moved between strings around the guitar fret neck. This approach is not really common for pop-rock guitarists and sometimes leads to unplayable intervals or chords. We've got a task from UG team to implement a logic which is more common. 
```
Here're main requirements of transposition of notes on Tab staff:
- Transposition on tab staff should NOT change strings, only frets
- If transposed note goes below the fret 0 it should be moved to one string lower (if no lower strings available negative red fret should be rendered, e.g. -2)
- If transposed note goes above the fret 24 it should be moved to one string higher (if no higher strings available red frets should be rendered, e.g. 26)
```

The option should be reproduced as a checkbox in the Transposition dialog with a text "Keep the same fingering for tabulature staves (where possible)". When checkbox is enabled system should keep the same fingering, when disabled - current transposition logic should be used.

Use the design for implementation
![Screenshot 2023-06-01 at 4 16 52 pm](https://github.com/musescore/MuseScore/assets/101250347/398c3354-fe34-4fa1-9481-f8959a5560e4)

### Problem to be solved

Additional, more common transposition approach will be implemented

**Comment by bkunda** [comment]
@mikekirin @abariska is this still a priority for you? 
If not, I'll move it out of the 4.4 project.

**Comment by abariska** [comment]
@bkunda it's not critical, could be definitely removed from 4.4

**Comment by MarcSabatella** [comment]
I agree it makes sense to include this in the transposition dialog.  But as far as priority goes, note that transposition via Up/Down already preserves string assignments, and I think most people writing for guitar tablature are accustomed to taking advantage f that feature.  However, it does not handle cases where the string *does* need to be changed, so unfortunately it is a bit limited.  It would be nice if that could use the same logic as proposed for the transpose dialog above.
